File: Chapter04/java/passbyvalue.java
public class passbyvalue {
    public static void main(String[] args) {
        int a = 1;
        int b = 2;
        System.out.println("Before swap, a = " + a);
        System.out.println("Before swap, b = " + b);

        swap(a, b);

        System.out.println("After swap, a = " + a);
        System.out.println("After swap, b = " + b);
    }

    private static void swap(int a, int b) {
        int temp;
        temp = a;
        a = b;
        b = temp;
        System.out.println("inside swap: a = " + a);
        System.out.println("inside swap: b = " + b);
    }

}

File: Chapter04/passbyvalue.py
def swap(num1, num2):
    # print(id(num1))
    # print(id(num2))
    temp = num1
    num1 = num2
    num2 = temp
    # print(id(num1))
    # print(id(num2))


a = 1
b = 2
# print(id(a))
# print(id(b))
print("Before swap: a = ", a)
print("Before swap: b = ", b)
swap(a, b)
print("After swap: a = ", a)
print("After swap: b = ", b)
# print(id(a))
# print(id(b))
